web/packages/teleport/src/components/FormLogin/FormIdentifierFirst.tsx

```diff
@@ -0,0 +1,275 @@
+import { AuthProvider } from 'shared/services';
+
+import ResourceService from 'teleport/services/resources';
+import { storageService } from 'teleport/services/storageService';
+
+import SSOButtonList from './SsoButtons';
+
+type Props = {
+  onLoginWithSso(provider: AuthProvider): void;
+  /**
+   * onUseLocalLogin is called to switch the view to the local login form.
+   */
+  onUseLocalLogin(): void;
+};
+
+export function FormIdentifierFirst({
+  onLoginWithSso,
+  onUseLocalLogin,
+}: Props) {
+  const [resourceService] = useState(() => new ResourceService());
+
+  const [rememberedUsername, setRememberedUsername] = useState<string>(
+    storageService.getRememberedSsoUsername()
+  );
+  const [username, setUsername] = useState<string>(rememberedUsername);
+  const [connectors, setConnectors] = useState<AuthProvider[]>([]);
+
+  useEffect(() => {
+    if (rememberedUsername) {
+      fetchMatchingConnectors(rememberedUsername);
+    }
+  }, [rememberedUsername]);
+
+  const [fetchAttempt, fetchMatchingConnectors] = useAsync(
+    useCallback(
+      async (username: string) => {
+        const connectors =
+          await resourceService.getUserMatchedAuthConnectors(username);
+        if (connectors.length === 0) {
+          if (rememberedUsername) {
+            // If we have a remembered username but no connectors, we clear the remembered username.
+            storageService.clearRememberedSsoUsername();
+            setRememberedUsername('');
+            setUsername('');
+            return;
+          }
+          throw new Error(`No SSO connectors found for user: ${username}`);
+        }
+        // If there isn't a remembered username, and there is only one matching connector, we take them straight to the IdP.
+        if (connectors.length === 1 && !rememberedUsername) {
+          onLoginWithSso(connectors[0]);
+          storageService.setRememberedSsoUsername(username);
+          setRememberedUsername(username);
+          return;
+        }
+        setRememberedUsername(username);
+        storageService.setRememberedSsoUsername(username);
+        return;
+      },
+      [username]
+    )
+  );
+
+  const onSubmitUsername = () => {
+    fetchMatchingConnectors(username.trim());
+  };
+
+  const onNotYou = () => {
+    storageService.clearRememberedSsoUsername();
+    setUsername('');
+    setRememberedUsername('');
+    setConnectors([]);
+  };
+
+  return (
+    <Flex flexDirection="column" alignItems="center" width="100%">
+      {fetchAttempt.status === 'error' && (
+        <Alerts.Danger>{fetchAttempt.statusText}</Alerts.Danger>
+      )}
+      {!rememberedUsername ? (
+        <>
+          <UsernamePrompt
+            onSubmitUsername={onSubmitUsername}
+            username={username}
+            setUsername={setUsername}
+            fetchAttempt={fetchAttempt}
+          />
+          <ButtonLink
+            mt={3}
+            onClick={onUseLocalLogin}
```

Comment:
will the `onUseLocalLogin` show both uname pass field and the SSO connector icons like we have today? or just takes you to local user login form?

Reply:
If identifier-first login is enabled, it will take you to local user login and not show the SSO connectors, unlike what we have today.